src/shared/dlt_config_file_parser.c

```diff
@@ -206,28 +196,35 @@ static int dlt_config_file_set_section_data(DltConfigFile *file, char *str1, cha
     /* copy data into structure */
     strncpy(&s->keys[key_number * DLT_CONFIG_FILE_ENTRY_MAX_LEN], str1, DLT_CONFIG_FILE_ENTRY_MAX_LEN);
 
-    item.key = strdup(str1);
-    item.data = strdup(str2);
-
-    /* check if already stored */
-    if (hsearch_r(item, FIND, &ret, &s->data) == 0)
+    if (s->list == NULL)
     {
-        if (hsearch_r(item, ENTER, &ret, &s->data) == 0)
+        s->list = malloc(sizeof(DltConfigKeyData));
+        if (s->list == NULL)
         {
-            free(item.key);
-            free(item.data);
-
-            dlt_log(LOG_ERR, "Cannot add data to hash table\n");
+            dlt_log(LOG_WARNING, "Could not allocate initial memory to list \n");
             return -1;
         }
+        tmp = &s->list;
     }
-    else /* already exist, ignore */
+    else
     {
-        char error_str[DLT_DAEMON_TEXTBUFSIZE] = {'\0'};
-        snprintf(error_str, DLT_DAEMON_TEXTBUFSIZE, "Key \"%s\" already exists! New data will be ignored\n", item.key);
-        dlt_log(LOG_WARNING, error_str);
-        return -1;
+        tmp = &s->list;
```

Comment:
missing a brief description what you are doing..
Looks like extending the existing list?